types/firestore.d.ts

```diff
@@ -122,9 +122,14 @@ declare namespace FirebaseFirestore {
      */
```

Comment:
BTW, I am inclined to stall development on this until we have settled on the final API. To make this easier, I updated the `firestore.d.ts` file with a new version that matches my understanding of where we ended up in the standup (minus naming)